### PR TITLE
Two level toolkit design with show more/less

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Maybe separate PR:
- fading background as per design for active one (just add to the effect)
- activation for Explore, no need yet but it's coming in search

See #376

### CODE REVIEW

File: src/documentation/common/ShowMoreButton.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    color="#262626"
+    variant="unstyled"
+    display="inline-flex"
+    justifyContent="flex-start"
+    size="sm"
+    opacity="0.6"
+    letterSpacing="0px"
+    p="0 12px"
+    textTransform="uppercase"
+    width="max-content"
```
Comment (microbit-matt-hillsdon) [inline src/documentation/common/ShowMoreButton.tsx:25]: To review, feels like there's a lot here.
The font also feels lighter than in XD.

File: src/documentation/ToolkitDocumentation/ToolkitDocumentation.tsx [hunk 2]
```diff
@@ -46,7 +53,7 @@ export const ToolkitDocumentation = ({ toolkit }: ToolkitProps) => {
   );
   return (
     <ActiveTooklitLevel
-      key={urlParam}
+      key={urlParam.length > 0 ? 0 : 1}
```
Comment (microbit-matt-hillsdon) [inline src/documentation/ToolkitDocumentation/ToolkitDocumentation.tsx:56]: Might be clearer for this to just use `anchor` now.

Comment (microbit-matt-hillsdon) [reply src/documentation/ToolkitDocumentation/ToolkitDocumentation.tsx]: Rewriting in terms of anchor here (and surrounding code here and in reference) improved things.

File: src/documentation/common/ShowMoreButton.tsx (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply src/documentation/common/ShowMoreButton.tsx]: I've dropped what I can here.
The font weight is the most significant different from the design here.

**Comment by microbit-matt-hillsdon** [comment]
Visually, I still don't like "SHOW MORE" but I think we should probably merge without worrying about that and then consider options (mostly because this is holding up other PRs).

**Comment by microbit-matt-hillsdon** [comment]
Spotted a spacing regression in Reference.
New:
image
Old:
image

**Comment by microbit-matt-hillsdon** [comment]
Preview URL is: 

https://review-python-editor-next.microbit.org/two-level-design-2/

**Review by microbit-robert — CHANGES_REQUESTED** [review]
- Scrolling to reference documentation from the link in the signature help is a little inconsistent. It always scrolls, but not necessarily to the correct place (e.g. for `display.show()`, I can see a 'show less' at the top of the scroll area).
- The link in signature help for anything in `audio` only takes you to references docs level 1, and not into the audio section for some reason.
- If not already inside the relevant second level of the reference toolkit, clicking the link in the signature help a second time after scrolling has finished, appears to cause a flash/glitch in the toolkit. E.g., `display.set_pixel()`

**Comment by microbit-matt-hillsdon** [comment]
There's an issue open about an issue with audio which might be related: #317

**Comment by microbit-matt-hillsdon** [comment]
> * Scrolling to reference documentation from the link in the signature help is a little inconsistent. It always scrolls, but not necessarily to the correct place (e.g. for `display.show()`, I can see a 'show less' at the top of the scroll area).

I think this is #325 in "show more" form. We don't uniquely identify the overload of scroll so we expand both which can cause the offset issue. Can you reproduce this with a method that doesn't have an overload?

> * The link in signature help for anything in `audio` only takes you to references docs level 1, and not into the audio section for some reason.

I'll look at this with #317.

> * If not already inside the relevant second level of the reference toolkit, clicking the link in the signature help a second time after scrolling has finished, appears to cause a flash/glitch in the toolkit. E.g., `display.set_pixel()`

New issue:

URLs are utterly messed up! E.g. ends with `%5Bobject%20Object%5D`. We need to unwrap Anchor.

File: src/documentation/reference/ReferenceToolkit.tsx [hunk 3]
```diff
@@ -3,109 +3,85 @@
...
-      : "back";
+  const [anchor, setAnchor] = useRouterParam(RouterParam.reference);
+  const handleNavigate = useCallback(
+    (id: string | undefined) => {
+      setAnchor(id ? { id } : undefined);
+    },
+    [setAnchor]
+  );
+  const direction = useAnimationDirection(anchor);
+  console.log({ anchor });
```
Comment (microbit-matt-hillsdon) [inline src/documentation/reference/ReferenceToolkit.tsx:36]: Unwanted debug:

```suggestion
```

**Comment by microbit-matt-hillsdon** [comment]
> > * Scrolling to reference documentation from the link in the signature help is a little inconsistent. It always scrolls, but not necessarily to the correct place (e.g. for `display.show()`, I can see a 'show less' at the top of the scroll area).
> 
> I think this is #325 in "show more" form. We don't uniquely identify the overload of scroll so we expand both which can cause the offset issue. Can you reproduce this with a method that doesn't have an overload?

We could and I've fixed it with a delay so the final height of the container accounts for the show-more expand before the scroll.

> > * The link in signature help for anything in `audio` only takes you to references docs level 1, and not into the audio section for some reason.
> 
> I'll look at this with #317.

Still leaving that to look at separately.

> > * If not already inside the relevant second level of the reference toolkit, clicking the link in the signature help a second time after scrolling has finished, appears to cause a flash/glitch in the toolkit. E.g., `display.set_pixel()`
> 
> I can't make this happen.

We reproduced this and I've finally fixed it in Slide.tsx (unintentional tree structure change causing component recreation).

> New issue:
> 
> URLs are utterly messed up! E.g. ends with `%5Bobject%20Object%5D`. We need to unwrap Anchor.

This is also fixed.

I think this PR is good to go. I'll address the fading background separately.

**Comment by microbit-matt-hillsdon** [comment]
> Spotted a spacing regression in Reference. New: image Old: image

This is also fixed.

**Comment by microbit-matt-hillsdon** [comment]
One issue to look at after merge:

- If there's a CM in the Collapse then it seems to fade out but not slide up, e.g. display.scroll. Clearer if you remove the other content in the Collapse. What's going on?

**Review by microbit-robert — APPROVED** [review]
LGTM

**Comment by microbit-matt-hillsdon** [comment]
> * If there's a CM in the Collapse then it seems to fade out but not slide up, e.g. display.scroll. Clearer if you remove the other content in the Collapse

Now #407

**Pull request merged**